### PR TITLE
[FIX] point_of_sale, pos_sms: traceback occurs while sending pos receipt

### PR DESCRIPTION
**Pull request by paan-odoo**
Before this commit:
==========
- There was a traceback error when sending receipts through SMS and WhatsApp.
- Basic receipts were not being added as email attachments when sending receipts through email.

After this commit:
==========
- The traceback error while sending receipts has been resolved, and the receipt sending flow is now seamless.
- Basic receipts will now be attached as email attachments when sending receipts through email.

task-4212901

Related:
- Enterprise: https://github.com/odoo/enterprise/pull/70891

### CODE REVIEW

File: addons/point_of_sale/views/pos_order_view.xml [hunk 1]
```diff
@@ -189,7 +189,7 @@
                                     <label for="email"/>
                                     <div class="o_row">
                                         <field name="email"/>
-                                        <button name="action_send_mail" icon="fa-envelope fa-lg" type="object" class="btn-link" title="email" invisible="not email"/>
+                                        <button name="action_send_mail" icon="fa-envelope fa-lg" type="object" class="btn-link" title="email" invisible="not email or context.get('hide_btns', False)"/>
```
Comment (vlst-odoo) [inline addons/point_of_sale/views/pos_order_view.xml:192]: why would you hide the button ?

Comment (paan-odoo) [reply addons/point_of_sale/views/pos_order_view.xml:192]: Hello @vlst-odoo 

With the click of that button, the Mail Composer opens to send marketing emails from the PoS order form view. This process requires mail dependency, which is not added to PoS, resulting in a dependency error.

Thanks!

image

Comment (vlst-odoo) [reply addons/point_of_sale/views/pos_order_view.xml:192]: normally i would suggest importing the assets. We were going to do that anyways: https://github.com/odoo/odoo/pull/178372/files#diff-ef73b6797b07294478f0a4f3956cd6243f9b972210fbe13c48554645f6c0f39eL134-L184

It's up to you

Comment (paan-odoo) [reply addons/point_of_sale/views/pos_order_view.xml]: Hello @vlst-odoo 

Changed the commit description.
Extra Fix- A case where the action is null, is handled in web_studio. (Affecting in restaurant appointment tour)

Thanks!

**Comment by paan-odoo** [comment]
Hello @vlst-odoo 

Changes:
- Removed the manifest file changes from the PR.

Could you please review this again?

Thanks

**Review by vlst-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**